tests/main/ubuntu-core-config-defaults-once/task.yaml

```diff
@@ -0,0 +1,117 @@
+summary: |
+   Test that configuration defaults are only applied once.
+
+# ubuntu-core-16: it is not yet possible to install snapd on core16
+systems: [ubuntu-core-18-*]
+
+environment:
+    GADGET_FILE: gadget-defaults.yaml
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB"/systemd.sh
+
+    systemctl stop snapd.service snapd.socket
+    rm -rf /var/lib/snapd/assertions/*
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+    unsquashfs -no-progress /var/lib/snapd/snaps/pc_*.snap
+
+    # Update the gadget config file
+    cat "$GADGET_FILE" >> squashfs-root/meta/gadget.yaml
+
+    mksquashfs squashfs-root pc_x1.snap -comp xz -no-fragments -no-progress
+    rm -rf squashfs-root
+    cp pc_x1.snap /var/lib/snapd/seed/snaps/
+    cp /var/lib/snapd/seed/seed.yaml seed.yaml.bak
+    python3 ./manip_seed.py /var/lib/snapd/seed/seed.yaml
+    cp "$TESTSLIB"/assertions/developer1.account /var/lib/snapd/seed/assertions
+    cp "$TESTSLIB"/assertions/developer1.account-key /var/lib/snapd/seed/assertions
+    cp "$TESTSLIB"/assertions/testrootorg-store.account-key /var/lib/snapd/seed/assertions
+
+    # kick first boot again
+    systemctl start snapd.service snapd.socket
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    echo "Undo the service disable"
+    rm -f /etc/ssh/sshd_not_to_be_run
+    for service in ssh rsyslog; do
+        systemctl unmask "$service.service" || true
+        systemctl enable "$service.service" || true
+        systemctl start "$service.service" || true
+    done
+
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB"/systemd.sh
+    systemctl stop snapd.service snapd.socket
+    rm -rf /var/lib/snapd/assertions/*
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+
+    if systemctl status snap-pc-x1.mount ; then
+       systemctl stop snap-pc-x1.mount
+       rm -f /etc/systemd/system/snap-pc-x1.mount
+       rm -f /etc/systemd/system/multi-user.target.wants/snap-pc-x1.mount
+       rm -f /var/lib/snapd/snaps/pc_x1.snap
+       systemctl daemon-reload
+    fi
+    rm /var/lib/snapd/seed/snaps/pc_x1.snap
+
+    cp seed.yaml.bak /var/lib/snapd/seed/seed.yaml
+    rm -f /var/lib/snapd/seed/assertions/developer1.account
+    rm -f /var/lib/snapd/seed/assertions/developer1.account-key
+    rm -f /var/lib/snapd/seed/assertions/testrootorg-store.account-key
+    # kick first boot again
+    systemctl start snapd.service snapd.socket
+    # wait for first boot to be done
+    while ! snap changes | grep -q "Done.*Initialize system state"; do sleep 1; done
```

Comment:
same suggestion as before, but it occurs to me that if there's some issue in the prepare code and we broke snapd in such a way that we can't restart it here, will this block forever? is there a timeout for restore sections in spread? /me hopes so :crossed_fingers: